### DISCUSSION TITLE
Code Quality Control

### DISCUSSION THREAD

**Discussion by mocsharp** [Project-MONAI/monai-deploy#64] [General]
SonarCloud.io & CodeCov.io are enabled on most of our MONAI Deploy projects for static code analysis.  We would like to get everyone's feedback on defining the metrics for code quality control.

The following screenshot shows the default metric value for each category.

![image](https://user-images.githubusercontent.com/4236590/163435542-192c921f-5c0a-44d3-a955-3d4a26b0e663.png)

**Comment by mocsharp** [comment]
Suggest lowering the coverage to 70% since some code may not be testable while the other can be tested through integration.

**Comment by ericspod** [comment]
There might be good reason why there's a lot of duplicate code in places, for example tutorials may have a lot of common setup code.

**Reply by MMelQin** [reply]
Also, current scans are set up for the whole repo, at least that is the case for App SDK including the example apps, while the released SDK pypi package only includes a portion of it, i.e. under monai/deploy, so applying the same metrics threshold, e.g. coverage and duplication, to the whole project is not desirable.

**Reply by mocsharp** [reply]
You may configure to exclude certain directories from scanning as described [here](https://docs.sonarcloud.io/advanced-setup/analysis-scope/#restrict-the-scope-of-coverage-detection).  

Edit: Forgot to mention this is per-project configuration

**Reply by MMelQin** [reply]
> You may configure to exclude certain directories from scanning as described [here](https://docs.sonarcloud.io/advanced-setup/analysis-scope/#restrict-the-scope-of-coverage-detection).

Yes, exclusion and preserving already resolved false positives/mitigated issues are par for the course. We'd need to have multiple setups for the same proi, one for the core deliverables/releases and one for everything else, and a common set of metrics threshold may not work across multiple subsystems for varying maturity/complexity.
